src/components/Search/SearchRouter/SearchRouterList.tsx

```diff
@@ -0,0 +1,149 @@
+import React, {forwardRef, useCallback} from 'react';
+import type {ForwardedRef} from 'react';
+import * as Expensicons from '@components/Icon/Expensicons';
+import type {SearchQueryJSON} from '@components/Search/types';
+import SelectionList from '@components/SelectionList';
+import SearchQueryListItem from '@components/SelectionList/Search/SearchQueryListItem';
+import type {SearchQueryItem, SearchQueryListItemProps} from '@components/SelectionList/Search/SearchQueryListItem';
+import type {SectionListDataType, SelectionListHandle, UserListItemProps} from '@components/SelectionList/types';
+import UserListItem from '@components/SelectionList/UserListItem';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import type {OptionData} from '@libs/ReportUtils';
+import * as SearchUtils from '@libs/SearchUtils';
+import * as Report from '@userActions/Report';
+import ROUTES from '@src/ROUTES';
+
+type ItemWithQuery = {
+    query: string;
+};
+
+type SearchRouterListProps = {
+    currentQuery: SearchQueryJSON | undefined;
+    reportForContextualSearch?: OptionData;
+    recentSearches: ItemWithQuery[] | undefined;
+    recentReports: OptionData[];
+    onSearchSubmit: (query: SearchQueryJSON | undefined) => void;
+    updateUserSearchQuery: (newSearchQuery: string) => void;
+    closeAndClearRouter: () => void;
+};
+
+function isSearchQueryListItem(listItem: UserListItemProps<OptionData> | SearchQueryListItemProps): listItem is SearchQueryListItemProps {
+    if ('singleIcon' in listItem.item && listItem.item.singleIcon && 'query' in listItem.item && !!listItem.item.query) {
+        return true;
+    }
+    return false;
+}
+
+function SearchRouterItem(props: UserListItemProps<OptionData> | SearchQueryListItemProps) {
```

Comment:
NAB might be good to separate this component into its own file.

Reply:
I was thinking about this too.
The argument for keeping `SearchRouterItem` component in `SearchRouterList` is that it's highly specific to search router list feature and I don't want it to be used anywhere else. When we keep this component here it's less likely that somebody will export it and use for something not related.